components/candy-machine/UpdateCandyMachine.tsx

```diff
@@ -178,22 +184,44 @@ const UpdateCandyMachine: FC<{
 
                 state.data.solTreasuryAccount = state.wallet
                 state.data.itemsRedeemed = state.itemsRedeemed
-                setErrorMessage('')
+                setError('')
                 return state.data
             } catch (err) {
-                setErrorMessage((err as Error).message)
+                setError((err as Error).message)
             }
             setIsLoading(false)
         }
     }
 
     useEffect(() => {
-        setErrorMessage('')
+        setError('')
         setIsLoading(false)
         fetchCandyMachine().then(setCandyMachineConfig)
         setIsLoading(false)
     }, [connection])
 
+    useEffect(() => {
```

Comment:
if we already have 
```
const { onChange, onSubmit, values, setValues } = useForm(updateCandyMachineV2, initialState)
```
why do we need this here? I somehow feel that this code is duplicated

Reply:
When the component is mounted, useForm is initialized before initialState and therefore the value state has outdated data. For example, when trying to update a candy machine, even though the price is 0,1 sol, it always gives an error that the price cannot be 0
By this, creates a new initialState when `candyMachineConfig` changes and update values state